skxray/correlation.py

```diff
@@ -0,0 +1,371 @@
+    future_intensity_norm at each level, symmetric normalization is used.
+
+    Parameters
+    ----------
+    buf : array
+        image data array to use for correlation
+
+    G : array
+        matrix of auto-correlation function without
+        normalizations
+
+    past_intensity_norm : array
+        matrix of past intensity normalizations
+
+    future_intensity_norm : array
+        matrix of future intensity normalizations
+
+    labels : array
+        labels of the required region of interests(roi's)
+
+    num_bufs : int, even
+        number of buffers(channels)
+
+    num_pixels : array
+        number of pixels in certain roi's
+        roi's, dimensions are : [number of roi's]X1
+
+    img_per_level : array
+        to track how many images processed in each level
+
+    level : int
+        the current multi-tau level
+
+    buf_no : int
+        the current buffer number
+
+    Notes
+    -----
+    :math ::
+        G   = <I(t)I(t + delay)>
+
+    :math ::
+        past_intensity_norm = <I(t)>
+
+    :math ::
+        future_intensity_norm = <I(t + delay)>
+
+    """
+    img_per_level[level] += 1
+
+    # in multi-tau correlation other than first level all other levels
+    #  have to do the half of the correlation
+    if level == 0:
+        i_min = 0
+    else:
+        i_min = num_bufs//2
+
+    for i in range(i_min, min(img_per_level[level], num_bufs)):
+        t_index = level*num_bufs/2 + i
+
+        delay_no = (buf_no - i) % num_bufs
+
+        past_img = buf[level, delay_no]
+        future_img = buf[level, buf_no]
+
+        #  get the matrix of auto-correlation function without normalizations
+        tmp_binned = (np.bincount(labels,
+                                  weights=np.ravel(past_img*future_img))[1:])
+        G[t_index] += ((tmp_binned/num_pixels - G[t_index])/
+                       (img_per_level[level] - i))
+
+        # get the matrix of past intensity normalizations
+        pi_binned = (np.bincount(labels,
+                                 weights=np.ravel(past_img))[1:])
+        past_intensity_norm[t_index] += ((pi_binned/num_pixels
+                                         - past_intensity_norm[t_index])/
+                                         (img_per_level[level] - i))
+
+        # get the matrix of future intensity normalizations
+        fi_binned = (np.bincount(labels,
+                                 weights=np.ravel(future_img))[1:])
+        future_intensity_norm[t_index] += ((fi_binned/num_pixels
+                                           - future_intensity_norm[t_index])/
+                                           (img_per_level[level] - i))
+
+    return None  # modifies arguments in place!
+
+
+def extract_label_indices(labels):
+    """
+    This will find the label's required region of interests (roi's),
+    number of roi's count the number of pixels in each roi's and pixels
+    list for the required roi's.
+
+    Parameters
+    ----------
+    labels : array
```

Comment:
Please rename the input variable for the same reason as above.  It makes debugging code you are not familiar with super annoying.